### PR TITLE
DOC: update the pandas.Series.str.strip docstring

### PR DESCRIPTION
**Pull request by Cheukting**
- [ ] closes #xxxx
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/strings.py [hunk 1]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
-    Series/Index from %(side)s. Equivalent to :meth:`str.%(method)s`.
+    Remove leading and trailing characters.
+
+    Strip whitespaces (including newlines) or a set of specified characters
+    from each string in the Series/Index from %(side)s.
+    Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    to_strip : str
```
Comment (WillAyd) [inline pandas/core/strings.py]: `to_strip : str, optional`

File: pandas/core/strings.py [hunk 2]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+        Specifying the set of characters to be removed.
```
Comment (WillAyd) [inline pandas/core/strings.py]: Let's just simplify this and say "The set of characters to be removed". It matches the standard Python string method so no need to be too wordy

File: pandas/core/strings.py [hunk 3]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+    --------
+    str.slice : Slice substrings from each element in the Series/Index
+
+    Examples
+    --------
+    Striping whitespaces for Series
+
+    >>> s = pd.Series([' ant', 'bee ', ' cat  '])
+
+    >>> s
```
Comment (WillAyd) [inline pandas/core/strings.py]: This is simple enough that you don't need to print the Series as is

File: pandas/core/strings.py [hunk 4]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
     -------
     stripped : Series/Index of objects
+
+    See Also
+    --------
+    str.slice : Slice substrings from each element in the Series/Index
+
+    Examples
+    --------
+    Striping whitespaces for Series
```
Comment (WillAyd) [inline pandas/core/strings.py]: "Striping" is a typo here and below, though these examples are simple enough that I don't think you need to explain them

File: pandas/core/strings.py [hunk 5]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+
+    >>> s.str.strip()
+    0    ant
+    1    bee
+    2    cat
+    dtype: object
+
+    Striping a set of characters for Index
+
+    >>> df = pd.DataFrame(index=['1.ant ','2._bee__','3. cat_'])
```
Comment (WillAyd) [inline pandas/core/strings.py]: Just create the index directly instead of creating a DataFrame

`index = pd.Index(['1.ant ']) ...`

File: pandas/core/strings.py [hunk 6]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+
+    >>> df.index
```
Comment (WillAyd) [inline pandas/core/strings.py]: Again simple enough you don't need to print

File: pandas/core/strings.py [hunk 7]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+    2     cat  
+    dtype: object
+
+    >>> s.str.strip()
+    0    ant
+    1    bee
+    2    cat
+    dtype: object
+
+    Striping a set of characters for Index
```
Comment (WillAyd) [inline pandas/core/strings.py]: Typo here, but as mentioned above these are basic enough that I personally don't think you need to explain in detail

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you post the results of the doc string validator?

**Comment by Cheukting** [comment]
###############################################################################

################################## Validation #################################

###############################################################################


Docstring for "pandas.Series.str.strip" correct. :)

File: pandas/core/strings.py [hunk 8]
```diff
@@ -2151,12 +2151,54 @@ def encode(self, encoding, errors="strict"):
...
+        All combinations of this set of characters will be stripped.
+        Default value is None, which means whaitspaces will be removed.
 
     Returns
     -------
     stripped : Series/Index of objects
+
+    See Also
+    --------
+    str.slice : Slice substrings from each element in the Series/Index
```
Comment (datapythonista) [inline pandas/core/strings.py]: `slice` doesn't seem very related to strip.

Not sure if there is anything else related, but we should have here `Series.str.strip`, `Series.str.lstrip` and `Series.str.rstrip`, so they link each other.

**Review by datapythonista — COMMENTED** [review]
Added some comments.

Also, #20628 was working on this too. I think your changes look better, but you can take a look at that PR and see if there is any idea that you can copy here.

File: pandas/core/strings.py [hunk 9]
```diff
@@ -2159,46 +2159,52 @@ def encode(self, encoding, errors="strict"):
...
-    2    cat
+    0    1. Ant.
+    1    2. Bee!
+    2    3. Cat?
     dtype: object
 
-    Stripping a set of characters for Index
-
-    >>> idx = pd.Index(['1.ant ','2._bee__','3. cat_'])
+    >>> s.str.strip('123.!? \n\t')
```
Comment (datapythonista) [inline pandas/core/strings.py]: Personally I find this example overcomplicated. I think something more realistic and simple could be:
- Illustrate the default `to_strip` parameter (whitespaces) with `rstrip` in a value like ` text    \n'
- `lstrip(to_strip='0')` for a value like `0010010`)
- And then you can show `strip(to_strip=' \n0')` with both values

**Comment by alsobay** [comment]
Hi @Cheukting, great work! As @datapythonista pointed out in #20897, it seems that our wires crossed during the sprint and we ended up working on the same thing. 

Seeing as this PR got in earlier, please feel free to grab anything from our PR that might help address Marc's notes, and we'll close ours. If you don't plan to refine this PR, please let us know and we'll continue on #20897.

**Comment by Cheukting** [comment]
Thanks @alsobay ! I really like your PR and I think I may borrow some ideas and put it here. Feel free to keep commenting and reviewing this PR as it is our collaboration effort :-)

File: pandas/core/strings.py (hunk 9 above)
Comment (Cheukting) [reply pandas/core/strings.py]: I  still wants to keep the 'Animal' theme in the example but I will simplify the `lstrip` and `rstrip` example

File: pandas/core/strings.py [hunk 10]
```diff
@@ -2151,12 +2151,60 @@ def encode(self, encoding, errors="strict"):
...
-    Series/Index from %(side)s. Equivalent to :meth:`str.%(method)s`.
+    Remove leading and trailing characters.
+
+    Strip whitespaces (including newlines) or a set of specified characters
+    from each string in the Series/Index from %(side)s.
+    Equivalent to :meth:`str.%(method)s`.
+
+    Parameters
+    ----------
+    to_strip : str or None, default None (whitespaces are removed)
```
Comment (jreback) [inline pandas/core/strings.py]: the coment about whitespaces are removed should be on the next line with a case, e.g.
If None then whitespaces are removed

File: pandas/core/strings.py [hunk 11]
```diff
@@ -2151,12 +2151,60 @@ def encode(self, encoding, errors="strict"):
...
+
+    See Also
+    --------
+    Series.str.strip : Remove leading and trailing characters in Series/Index
+    Series.str.lstrip : Remove leading characters in Series/Index
+    Series.str.rstrip : Remove trailing characters in Series/Index
+
+    Examples
+    --------
+    >>> s = pd.Series(['1. Ant.  ', '2. Bee!\n', '3. Cat?\t'])
```
Comment (jreback) [inline pandas/core/strings.py]: can you add a np.nan in the example (in place of an entry or in addition is ok)

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mroeschke** [comment]
Looks great! Will merge on green.

**Comment by jreback** [comment]
@mroeschke merge away

**Review by jreback — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Cheukting!
